### DISCUSSION TITLE
What is the difference between Coin and FungibleAsset?

### DISCUSSION THREAD

**Discussion by 0xmaayan** [aptos-labs/aptos-developer-discussions#208] [Questions] [answered]
### Discord user ID

casimirknight

### Describe your question in detail.

hey i have a doubt 
```
transferFungibleAsset(args: {
        sender: Account;
        fungibleAssetMetadataAddress: AccountAddressInput;
        recipient: AccountAddressInput;
        amount: AnyNumber;
        options?: InputGenerateTransactionOptions;
    }): Promise<SimpleTransaction>;
```

and 
```
transferCoinTransaction(args: {
        sender: AccountAddressInput;
        recipient: AccountAddressInput;
        amount: AnyNumber;
        coinType?: MoveStructId;
        options?: InputGenerateTransactionOptions;
    }): Promise<SimpleTransaction>;
```
1. What is the acutal difference between them?
2. Can we use both to transfer USDT or USDC?
3. what is `fungibleAssetMetadataAddress`? for example `0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDC` is USDC coin type then is 
`0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa` the `fungibleAssetMetadataAddress`

4. If no then how can i find it?

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

N/A

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment, marked as answer]
> 1. What is the actual difference between them?

[Coin](https://aptos.dev/standards/aptos-coin) is the original standard for fungible tokens, similar to ERC-20 in the Ethereum space.  It requires passing generic arguments to identify the coins, which is good in some cases, and less so in others.  It uses a `CoinStore` resource to store coins in an account, but can be moved anywhere (and not dropped).

[Fungible Asset](https://aptos.dev/standards/fungible-asset) is the newer standard for fungible tokens, that allows for anything to be fungible.  More specifically, it uses an [Object](https://aptos.dev/standards/aptos-object) to store in a Primary store.  Advantages here is that it does not require generics, instead it requires reference to the fungible asset `Metadata` object.

> 2. Can I use them to transfer stablecoins e.g. (USDC / USDT)

These don't specify stablecoins directly associated today.  You can take a look at bridged stablecoins from LayerZero, Wormhole, Celer or others.  Most of these are today as a Coin standard, but we are planning to have coins to be able to be migrated to fungible asset seamlessly.

> 3. What is `FungibleAssetMetadata`?
FungibleAssetMetadata describes the fungible asset name and properties.  Here is an example for the CELLANA token: https://explorer.aptoslabs.com/object/0x2ebb2ccac5e027a87fa0e2e5f656a3a4238d6a48d93ec9b610d570fc0aa0df12/resources?network=mainnet 

For coins, like your example `0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDC` there is instead a `CoinInfo` resource stored at that address see https://explorer.aptoslabs.com/account/0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa/resources?network=mainnet
